main/FatesRestartInterfaceMod.F90

```diff
@@ -1124,25 +1133,35 @@ subroutine define_restart_vars(this, initialize_variables)
     
     ! Only register satellite phenology related restart variables if it is turned on!
```

Comment:
Minor cleanup; since we use `hlm_use_sp` in a bunch of places still, I think moving this up to be included in a general comment would be ideal.  Or we can strike it altogether since it's not relevant in this specific place in the code anymore.

Reply:
discussed with @glemieux will incorporate into future PR